### PR TITLE
Move NameGenerator to util

### PR DESCRIPTION
**Pull request by PeterChen13579**
Fixes for #876

### CODE REVIEW

File: tests/common/util/NameGenerator.hpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+namespace tests::util {
+
+struct NameGenerator {
+    inline static auto
+    GenerateName()
+    {
+        return [](auto const& info) { return info.param.testName; };
+    }
+};
```
Comment (kuznetsss) [inline tests/common/util/NameGenerator.hpp]: You could do it shorter:
```cpp
static const auto NameGenerator = [](auto const& info) { return info.param.testName; };
```
And then you also will not need to put `()` when using it.

**Review by kuznetsss — DISMISSED** [review]
👍

File: tests/common/util/NameGenerator.hpp [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+#pragma once
+
+#include <string>
+
+namespace tests::util {
+
+static auto const NameGenerator = [](auto const& info) { return info.param.testName; };
+
+}  // namespace tests::util
```
Comment (godexsoft) [inline tests/common/util/NameGenerator.hpp]: Please set your IDE up to automatically add `\n` at end of files on save 👍

**Review by godexsoft — CHANGES_REQUESTED** [review]
Looks good 💯 
Just one thing - let's set your IDE up so github is happy 🥇

File: tests/common/util/NameGenerator.hpp (hunk 2 above)
Comment (kuznetsss) [reply tests/common/util/NameGenerator.hpp]: Why do we need to have an empty line at the end of each file?

If we think it is useful, we could force it on project level by adjusting clang-tidy's config to add new line on formatting.

Comment (godexsoft) [reply tests/common/util/NameGenerator.hpp]: Yeah, we probably should set it up then. It's not "useful" but imo it's always better to satisfy github so it does not display this red icon at the bottom of each file in the diffs.

Comment (kuznetsss) [reply tests/common/util/NameGenerator.hpp]: The red icon just indicates the end of file. 
https://stackoverflow.com/questions/77201426/what-does-a-red-icon-with-a-hyphen-mean-in-a-github-pr

Comment (kuznetsss) [reply tests/common/util/NameGenerator.hpp]: But I don't mind if we turn on this rule for clang-tidy. But it should be fixed separately and maybe even the file could be fixed automatically by clang-tidy.

Comment (godexsoft) [reply tests/common/util/NameGenerator.hpp]: That's just misinformation. Even your link says "no newline at end of file" in the accepted answer.
In my experience, the icon only ever appears if the newline is missing at the end of the file. I'm for consistency - the rest of the codebase (hopefully) follows the newline rule. And if we can enforce it with clang-format/tidy then we should do that as well 👍

Comment (kuznetsss) [reply tests/common/util/NameGenerator.hpp]: I created an issue #1434

**Review by godexsoft — APPROVED** [review]

**Pull request merged**